api.planx.uk/inviteToPay/createPaymentSendEvents.ts

```diff
@@ -0,0 +1,68 @@
+import { NextFunction, Request, Response } from 'express';
+import { createScheduledEvent } from "../hasura/metadata";
+import { _admin as $admin } from '../client';
+import { getMostRecentPublishedFlow } from '../helpers';
+
+type Destination = "bops" | "uniform" | "email";
+
+interface CombinedResponse {
+  bops?: Record<string, string>;
+  uniform?: Record<string, string>;
+  email?: Record<string, string>;
+}
+
+// Create "One-off Scheduled Events" in Hasura when a payment request is paid
+const createPaymentSendEvents = async (req: Request, res: Response, next: NextFunction): 
+  Promise<NextFunction | Response | void> => {
+  try {
+    const now = new Date();
+    const combinedResponse: CombinedResponse = {};
+
+    // TODO Get this session's published flow data, find it's Send component, determine which "destinations" we need to queue up events for
+    const session = await $admin.getSessionById(req.params.sessionId);
+    const publishedFlowData = await getMostRecentPublishedFlow(session.flowId);
+    const destinations: Destination[] = [];
+
+    // TODO Determine which local authority it should send to based on team & destination    
+
+    const eventPayload = { sessionId: req.params.sessionId };
+    if ("bops" in destinations) {
+      const bopsEvent = await createScheduledEvent({
+        webhook: `{{HASURA_PLANX_API_URL}}/bops/${req.body.bops.localAuthority}`,
```

Comment:
Not a strong opinion, but there might be scope here for simplification here by dropping the payload and just having a route of  something like `{{HASURA_PLANX_API_URL}}/submit/${sessionId}/bops/${localAuthority}`

Reply:
Yes - on the same page here - but I think it's a v2 enhancement if that's alright! I'd actually like to get to the point where each submission endpoint _only_ requires the `:sessionId` param and `:localAuthority` logic is moved to each respective submission domain endpoint rather than at this outer level (eg `createPaymentSendEvents` and in frontend `Send` component model) like now.

Reply:
Totally! No issues here 😄